ci: build: add job to build the example and upload to GitHub pages

hnez opened the pull request:
A build job can be a valuable resource for newcomers to a project to learn how to build the software.
Add a build job as an example of how to build a werkbank project and as an integration test to check if everything still compiles.
Also deploy the generated page to GitHub pages.

The deployment is conditional on two things:

  - `github.ref == 'refs/heads/main'`

    This ensures that deployment is only attempted for builds that run for the main branch, e.g. not pull requests or work on devel branches.

    See the [`github`][1] context documentation for reference.

  - `vars.DEPLOY_PAGES`

    A custom repository-wide [configuration variable][2].

    This ensures that the deployment is, by default, not attempted for forks (because the variable will not be set).

    The pages deployment requires a bit of configuration and would fail on a new fork, making the user experience less than ideal.

    Contributors that _do_ want to try the pages deployment can however set the variable in their fork and get working deployments that way. This makes it superior to e.g.: `github.repository == 'neusta-mobile-solutions-gmbh/werkbank'`.

[1]: https://docs.github.com/en/actions/reference/accessing-contextual-information-about-workflow-runs#github-context
[2]: https://docs.github.com/en/actions/how-tos/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository

## Required setup

Add a `DEPLOY_PAGES` variable in the repository settings and set it to a true-ish value:

![new_variable](https://github.com/user-attachments/assets/0470b7aa-a631-46f5-b688-46d4e88fc897)
![variable_overview](https://github.com/user-attachments/assets/047cf304-b673-4dcf-937b-d60cc95a377e)

Set the GitHub Pages build and deployment source to "GitHub Actions":

![pages_source](https://github.com/user-attachments/assets/3a64b75f-3c53-4a4b-b4a9-00b490058551)

With this the build and deploy jobs should run successfully (when merged into main), as seen [here](https://github.com/hnez/werkbank/actions/runs/16045129461) and the resulting page should be deployed to GitHub pages, as seen here: [hnez.github.io/werkbank](https://hnez.github.io/werkbank/).

LenWdk reviewed (commented):
Hey there. 

Thank you for contributing again. It look like you are ignoring the SDK Version mentioned in .fvmrc by using this container. Is that intended?

hnez commented:
> Thank you for contributing again. It look like you are ignoring the SDK Version mentioned in .fvmrc by using this container. Is that intended?

To be quite honest I didn't even know that the `.fvmrc` was even a thing. I've just updated the PR to use the same flutter SDK version as specified in the `.fvmrc` and also added a check to prevent the workflow file and the `.fvmrc` from drifting apart.

I will do the same for my other PRs at #16 and #17.

This also makes the scheduled builds quite useless, so I am going to remove those (I already did so here).

LenWdk commented:
That doesnt seem right. I mean it may works for now, but I would expect fvm to be used to install the matching flutter sdk. Maybe this is needed?  https://github.com/marketplace/actions/flutter-fvm-config-action

And consider adding `--release` to the build command

hnez commented:
> Thank you again @hnez. This has saved me a lot of time researching about GitHub Actions :D

You are welcome. I am quite happy to actually see it work :tada:!
